packages/flutter/lib/src/material/two_level_list.dart

```diff
@@ -0,0 +1,146 @@
+      )
+    );
+  }
+}
+
+class TwoLevelSublist extends StatefulComponent {
+  TwoLevelSublist({ Key key, this.left, this.center, this.children }) : super(key: key);
+
+  final Widget left;
+  final Widget center;
+  final List<Widget> children;
+
+  _TwoLevelSublistState createState() => new _TwoLevelSublistState();
+}
+
+class _TwoLevelSublistState extends State<TwoLevelSublist> {
+  final AnimationController controller = new AnimationController(duration: _kExpand);
+  bool isExpanded = false;
+
+  void initState() {
+    super.initState();
+    isExpanded = PageStorage.of(context)?.readState(context) ?? false;
+    if (isExpanded)
+      controller.value = 1.0;
+  }
+
+  void handleOnTap() {
+    setState(() {
+      isExpanded = !isExpanded;
+      if (isExpanded)
+        controller.forward();
+      else
+        controller.reverse();
+      PageStorage.of(context)?.writeState(context, isExpanded);
+    });
+  }
+
+  Widget buildList(BuildContext context, Widget child) {
+    final ThemeData theme = Theme.of(context);
+
+    final easeOutAnimation = new CurvedAnimation(parent: controller, curve: Curves.easeOut);
+    final easeInAnimation = new CurvedAnimation(parent: controller, curve: Curves.easeIn);
+
+    final ColorTween borderColor = new ColorTween(begin: Colors.transparent, end: theme.dividerColor);
+    final Tween<double> heightFactor = new Tween<double>(begin: 0.0, end: 1.0);
+    final ColorTween headerColor = new ColorTween(begin: theme.text.body1.color, end: theme.accentColor);
+    final ColorTween iconColor = new ColorTween(begin: theme.unselectedColor, end: theme.accentColor);
+    final Animation<double> iconTurns = new Tween<double>(begin: 0.0, end: 0.5).animate(easeInAnimation);
+
+    return new Container(
+      decoration: new BoxDecoration(
+        border: new Border(
+          top: new BorderSide(color: borderColor.evaluate(easeOutAnimation)),
+          bottom: new BorderSide(color: borderColor.evaluate(easeOutAnimation))
+        )
+      ),
+      child: new Column(
+        children: <Widget>[
+          new TwoLevelListItem(
+            onTap: handleOnTap,
+            left: config.left,
+            center: new DefaultTextStyle(
+              style: theme.text.body1.copyWith(color: headerColor.evaluate(easeInAnimation)),
+              child: config.center
+            ),
+            right: new RotationTransition(
+              turns: iconTurns,
+              child: new Icon(
+                icon: 'navigation/expand_more',
+                colorFilter: new ColorFilter.mode(iconColor.evaluate(easeInAnimation), TransferMode.srcATop)
+              )
+            )
+          ),
+          new ClipRect(
+            child: new Align(
+              heightFactor: heightFactor.evaluate(easeInAnimation),
+              child: new Column(children: config.children)
+            )
+          )
+        ]
+      )
+    );
+  }
+
+  Widget build(BuildContext context) {
+    return new AnimatedBuilder(
```

Comment:
There's no need to set up all the tweens and and the curve animation every frame.  You can just set those up once and re-use them every frame of the animation.

The way I would recommend doing this is creating all the objects in `initState` (along with the _controller).  Then, in this `build` function, I'd grab the ThemeData and adjust the begin/end values as appropriate for the theme.  That way you're not grabbing the theme data every frame either.  The `buildList` function (which should be private) can then just read the current values.